### PR TITLE
Implement sign/verify requests/responses

### PR DESCRIPTION
**Pull request by rpiasetskyi**
* Implement sign/verify requests/responses
* Move ECC key pair generation from worker to crypto module

### CODE REVIEW

File: heimlig/src/crypto/ecdsa.rs [hunk 1]
```diff
@@ -1,60 +1,630 @@
...
+        /// # Errors
+        ///
+        /// The function returns an error if:
+        /// * `InvalidSignatureSize`: The length of the `signature` is not `
+        #[doc=$signature_size_str]
+        /// ` bytes.
+        /// * `InvalidDigest`: The length of the `digest` is not `
+        #[doc=$digest_size_str]
+        /// ` bytes.
+        /// * `InvalidPrivateKey`: The length of the `private_key` is not `
```
Comment (inorick) [inline heimlig/src/crypto/ecdsa.rs:234]: Should we rename our size errors to `InvalidDigestSize` and `InvalidPrivateKey` to stay consistent with `InvalidSignatureSize`?

File: heimlig/src/crypto/ecdsa.rs [hunk 2]
```diff
@@ -1,60 +1,630 @@
...
+
+                let entropy = rng::test::TestEntropySource::default();
+                let mut rng = rng::Rng::new(entropy, None);
+                let (private_key, public_key) = $generate_key_pair(&mut rng);
+
+                let digest =
+                    <$curve as DigestPrimitive>::Digest::new_with_prefix(MESSAGE).finalize();
+
+                // Invalid private key size.
+                for size in [0, 1, 16, 32, 48, 64] {
```
Comment (inorick) [inline heimlig/src/crypto/ecdsa.rs:511]: Why are "0" and "1" skipped?

File: heimlig/src/hsm/workers/ecc_worker.rs [hunk 3]
```diff
@@ -70,94 +125,337 @@ impl<
...
+    mut public_key: [u8; M],
+    private_key_bytes: &'a mut [u8],
+    public_key_bytes: &'a mut [u8],
+) -> (&'a [u8], &'a [u8]) {
+    let private_key_result = &mut private_key_bytes[..private_key.len()];
+    private_key_result.copy_from_slice(&private_key);
+    let public_key_result = &mut public_key_bytes[..public_key.len()];
+    public_key_result.copy_from_slice(&public_key);
+
+    private_key.zeroize();
```
Comment (inorick) [inline heimlig/src/hsm/workers/ecc_worker.rs:457]: As it is used now, the mutable input parameters are wrapped into `Zeroize` on the caller site anyway so we zeroize a second time here, right? If so, why do we have `move_key_pair` and not just copy from key to key on the caller site and let `Zeroize` do the cleanup?

File: heimlig/tests/api_core_worker_communication.rs [hunk 4]
```diff
@@ -706,6 +711,55 @@ mod tests {
         };
         assert_eq!(request_id, org_request_id);
         assert_eq!(private_key.len(), KEY_INFO.ty.private_key_size()); // Large buffer was only used partially
+
+        // Sign
```
[the remark this answers is not in the document]
Comment (rpiasetskyi) [reply heimlig/tests/api_core_worker_communication.rs]: Done.

File: heimlig/src/crypto/ecdsa.rs [hunk 5]
```diff
@@ -1,60 +1,630 @@
...
-    let key: VerifyingKey<C> = key.into();
-    key.verify(message, signature).is_ok()
+    let secret_key = SecretKey::<C>::random(rng);
+    let public_key = secret_key.public_key();
+
+    let encoded_point = public_key.to_encoded_point(false);
+
+    (
+        secret_key.to_bytes(),
+        PublicKeyBytes::<C>::clone_from_slice(&encoded_point.as_bytes()[1..]),
```
[the remark this answers is not in the document]
Comment (rpiasetskyi) [reply heimlig/src/crypto/ecdsa.rs:141]: Done.

File: heimlig/src/crypto/ecdsa.rs (hunk 1 above)
Comment (rpiasetskyi) [reply heimlig/src/crypto/ecdsa.rs:238]: Let me create it as a separate PR to fix it also for `ed25519`.

File: heimlig/src/hsm/workers/ecc_worker.rs (hunk 3 above)
Comment (rpiasetskyi) [reply heimlig/src/hsm/workers/ecc_worker.rs:457]: As discussed offline, these explicit `zeroize()` calls are used for temporary array variables.

**Review by inorick — APPROVED** [review]

**Pull request merged**